Assets/Scripts/WorldEngine/Manager.cs

```diff
@@ -983,9 +997,22 @@ public static void ExportMapTextureToFileAsync(string path, Rect uvRect, Progres
 
         Debug.Log("Trying to export world map to .png file: " + Path.GetFileName(path));
 
+        // Launch the thread job that will handle the async export task
         ThreadPool.QueueUserWorkItem(state =>
         {
-            ExportMapTextureToFile(path, uvRect);
+            try
+            {
+                ExportMapTextureToFile(path, uvRect);
+            }
+            catch (System.Exception e)
+            {
+                // To display the exception on screen we need to queue a task on the rendering engine's main thread
+                // and rethrow the exception inside that task
+                EnqueueTaskAndWait(() =>
+                {
+                    throw new System.Exception("Unhandled exception in ExportMapTextureToFile with path: " + path, e);
+                });
+            }
```

Comment:
Good addition to error handling.

Reply:
Yeah, I did the same for the other async tasks a while ago but I missed this particular case. Now it's taken care of.